resolwe_bio/processes/microarray/methylation_arrays.py

```diff
@@ -0,0 +1,145 @@
+"""Process methylation array data."""
+import re
+
+from pathlib import Path
+from shutil import copy2
+
+import pandas as pd
+
+from plumbum import TEE
+
+from resolwe.process import (
+    SchedulingClass,
+    Persistence,
+    Cmd,
+    StringField,
+    FileField,
+    DataField,
+)
+from resolwe_bio.process.runtime import ProcessBio
+
+
+class MethylationArraySesame(ProcessBio):
+    """Illumina methylation array SeSAMe process
+
+    Implemented SeSAMe method for analyzing methylation array from
+    Illumina. For more information on the pipeline, please see
+    https://www.bioconductor.org/packages/release/bioc/html/sesame.html
+
+    This process will input IDAT methylation array files and
+    produce two results. One is the quality control file with
+    some basic statistics, such as mean beta, fraction of
+    (un)methylated, GCT, predicted ethnicity, gender and age.
+    Expression file holds betas, mvals and pvals for probe ids.
+    """
+
+    slug = "methylation-array-sesame"
+    name = "Methylation analysis (SeSAMe)"
+    process_type = "data:methylation:sesame"
+    version = "1.0.0"
+    category = "Methylation arrays"
+    data_name = 'SeSAMe array ({{ idat_file.red_channel.file|default("?") }})'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {"type": "sample"}
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "public.ecr.aws/s4q6j6e8/resolwebio/methylation_arrays:1.0.0"
+            }
+        },
+        "resources": {
+            "cores": 8,
+            "memory": 16384,
+            "network": True,
+        },
+    }
+
+    class Input:
+        """Input fields to process MethylationArraySesame."""
+
+        idat_file = DataField(
+            data_type="methylationarray:idat",
+            label="Illumina methylation array IDAT file",
+            description="Illumina methylation array BeadChip raw IDAT file.",
+        )
+        manifest = StringField(
```

Comment:
Idat file holds the species/platform information. Could we remove this input, and map the species/platform -> manifest info within the process body?

Reply:
Oh, I see that there is hg19/hg38 choice also, so there is no real benefit of this.